In [1]:
import pandas as pd
import numpy as np

In [2]:
# Loading Data
data = pd.read_csv('housing.csv')

In [8]:
data.head(10)

longitude  latitude  housing_median_age  total_rooms  total_bedrooms  \
0    -122.23     37.88                41.0        880.0           129.0   
1    -122.22     37.86                21.0       7099.0          1106.0   
2    -122.24     37.85                52.0       1467.0           190.0   
3    -122.25     37.85                52.0       1274.0           235.0   
4    -122.25     37.85                52.0       1627.0           280.0   
5    -122.25     37.85                52.0        919.0           213.0   
6    -122.25     37.84                52.0       2535.0           489.0   
7    -122.25     37.84                52.0       3104.0           687.0   
8    -122.26     37.84                42.0       2555.0           665.0   
9    -122.25     37.84                52.0       3549.0           707.0   

   population  households  median_income  median_house_value ocean_proximity  
0       322.0       126.0         8.3252            452600.0        NEAR BAY  
1      2401.0      1138.0         8.3014            358500.0        NEAR BAY  
2       496.0       177.0         7.2574            352100.0        NEAR BAY  
3       558.0       219.0         5.6431            341300.0        NEAR BAY  
4       565.0       259.0         3.8462            342200.0        NEAR BAY  
5       413.0       193.0         4.0368            269700.0        NEAR BAY  
6      1094.0       514.0         3.6591            299200.0        NEAR BAY  
7      1157.0       647.0         3.1200            241400.0        NEAR BAY  
8      1206.0       595.0         2.0804            226700.0        NEAR BAY  
9      1551.0       714.0         3.6912            261100.0        NEAR BAY

In [9]:
data.describe()

longitude      latitude  housing_median_age   total_rooms  \
count  20640.000000  20640.000000        20640.000000  20640.000000   
mean    -119.569704     35.631861           28.639486   2635.763081   
std        2.003532      2.135952           12.585558   2181.615252   
min     -124.350000     32.540000            1.000000      2.000000   
25%     -121.800000     33.930000           18.000000   1447.750000   
50%     -118.490000     34.260000           29.000000   2127.000000   
75%     -118.010000     37.710000           37.000000   3148.000000   
max     -114.310000     41.950000           52.000000  39320.000000   

       total_bedrooms    population    households  median_income  \
count    20433.000000  20640.000000  20640.000000   20640.000000   
mean       537.870553   1425.476744    499.539680       3.870671   
std        421.385070   1132.462122    382.329753       1.899822   
min          1.000000      3.000000      1.000000       0.499900   
25%        296.000000    787.000000    280.000000       2.563400   
50%        435.000000   1166.000000    409.000000       3.534800   
75%        647.000000   1725.000000    605.000000       4.743250   
max       6445.000000  35682.000000   6082.000000      15.000100   

       median_house_value  
count        20640.000000  
mean        206855.816909  
std         115395.615874  
min          14999.000000  
25%         119600.000000  
50%         179700.000000  
75%         264725.000000  
max         500001.000000

### Here we observe that latitude has the minimum value of 32.54000 and housing_median_age has the minimum value of 1.0000, so we will have to scale the data accordingly.

### And if we didn't then there will be more weightage given to the latitude term while doing calcultions.

### There are mainly two types of scaling techniques namely min max scaling and Standard Scaling or normalisation.

In [7]:
from sklearn.preprocessing import MinMaxScaler

In [10]:
scaler=MinMaxScaler()

In [11]:
scaler.fit(data) #Will not work because there is a column containg string values

ValueError: could not convert string to float: 'NEAR BAY'

In [12]:
data=data.drop(data.columns[[9]],axis=1)

In [13]:
data.head()

longitude  latitude  housing_median_age  total_rooms  total_bedrooms  \
0    -122.23     37.88                41.0        880.0           129.0   
1    -122.22     37.86                21.0       7099.0          1106.0   
2    -122.24     37.85                52.0       1467.0           190.0   
3    -122.25     37.85                52.0       1274.0           235.0   
4    -122.25     37.85                52.0       1627.0           280.0   

   population  households  median_income  median_house_value  
0       322.0       126.0         8.3252            452600.0  
1      2401.0      1138.0         8.3014            358500.0  
2       496.0       177.0         7.2574            352100.0  
3       558.0       219.0         5.6431            341300.0  
4       565.0       259.0         3.8462            342200.0

In [14]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20640 entries, 0 to 20639
Data columns (total 9 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   longitude           20640 non-null  float64
 1   latitude            20640 non-null  float64
 2   housing_median_age  20640 non-null  float64
 3   total_rooms         20640 non-null  float64
 4   total_bedrooms      20433 non-null  float64
 5   population          20640 non-null  float64
 6   households          20640 non-null  float64
 7   median_income       20640 non-null  float64
 8   median_house_value  20640 non-null  float64
dtypes: float64(9)
memory usage: 1.4 MB


In [15]:
data=data.dropna(axis=0)

In [17]:
scaler=MinMaxScaler()
scaler.fit(data)

MinMaxScaler()

In [18]:
scaledData=pd.DataFrame(data=scaler.transform(data),columns=data.columns,index=data.index)
scaledData

longitude  latitude  housing_median_age  total_rooms  total_bedrooms  \
0       0.211155  0.567481            0.784314     0.022331        0.019863   
1       0.212151  0.565356            0.392157     0.180503        0.171477   
2       0.210159  0.564293            1.000000     0.037260        0.029330   
3       0.209163  0.564293            1.000000     0.032352        0.036313   
4       0.209163  0.564293            1.000000     0.041330        0.043296   
...          ...       ...                 ...          ...             ...   
20635   0.324701  0.737513            0.470588     0.042296        0.057883   
20636   0.312749  0.738576            0.333333     0.017676        0.023122   
20637   0.311753  0.732200            0.313725     0.057277        0.075109   
20638   0.301793  0.732200            0.333333     0.047256        0.063315   
20639   0.309761  0.725824            0.294118     0.070782        0.095438   

       population  households  median_income  median_house_value  
0        0.008941    0.020556       0.539668            0.902266  
1        0.067210    0.186976       0.538027            0.708247  
2        0.013818    0.028943       0.466028            0.695051  
3        0.015555    0.035849       0.354699            0.672783  
4        0.015752    0.042427       0.230776            0.674638  
...           ...         ...            ...                 ...  
20635    0.023599    0.054103       0.073130            0.130105  
20636    0.009894    0.018582       0.141853            0.128043  
20637    0.028140    0.071041       0.082764            0.159383  
20638    0.020684    0.057227       0.094295            0.143713  
20639    0.038790    0.086992       0.130253            0.153403  

[20433 rows x 9 columns]